migrate test-infra to testify for `br/pkg/lightning/tikv`

@YuJuncen @kennytm this test isn't running now. I tried to migrate it and re-enabled the test but got failures.

Could you please give it a look on how to fix?

<details>
<summary>my patch</summary>

```diff
diff --git a/br/pkg/lightning/tikv/tikv_test.go b/br/pkg/lightning/tikv/tikv_test.go
index f74b20c5c..bf29088d0 100644
--- a/br/pkg/lightning/tikv/tikv_test.go
+++ b/br/pkg/lightning/tikv/tikv_test.go
@@ -1,3 +1,17 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
 package tikv_test
 
 import (
@@ -9,18 +23,15 @@ import (
 	"net/url"
 	"sort"
 	"sync"
+	"testing"
 
 	"github.com/coreos/go-semver/semver"
-	. "github.com/pingcap/check"
 	"github.com/pingcap/kvproto/pkg/import_sstpb"
 	"github.com/pingcap/tidb/br/pkg/lightning/common"
 	kv "github.com/pingcap/tidb/br/pkg/lightning/tikv"
+	"github.com/stretchr/testify/require"
 )
 
-type tikvSuite struct{}
-
-var _ = Suite(&tikvSuite{})
-
 var (
 	// Samples from importer backend for testing the Check***Version functions.
 	// No need keep these versions in sync.
@@ -30,7 +41,7 @@ var (
 	requiredMaxTiKVVersion = *semver.New("6.0.0")
 )
 
-func (s *tikvSuite) TestForAllStores(c *C) {
+func TestForAllStores(t *testing.T) {
 	server := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
 		_, err := w.Write([]byte(`
 			{
@@ -84,7 +95,7 @@ func (s *tikvSuite) TestForAllStores(c *C) {
 				]
 			}
 		`))
-		c.Assert(err, IsNil)
+		require.NoError(t, err)
 	}))
 	defer server.Close()
 
@@ -100,11 +111,10 @@ func (s *tikvSuite) TestForAllStores(c *C) {
 		allStoresLock.Unlock()
 		return nil
 	})
-	c.Assert(err, IsNil)
+	require.NoError(t, err)
 
 	sort.Slice(allStores, func(i, j int) bool { return allStores[i].Address < allStores[j].Address })
-
-	c.Assert(allStores, DeepEquals, []*kv.Store{
+	require.Equal(t, []*kv.Store{
 		{
 			Address: "127.0.0.1:20160",
 			Version: "3.0.0-beta.1",
@@ -125,10 +135,10 @@ func (s *tikvSuite) TestForAllStores(c *C) {
 			Version: "3.0.1",
 			State:   kv.StoreStateOffline,
 		},
-	})
+	}, allStores)
 }
 
-func (s *tikvSuite) TestFetchModeFromMetrics(c *C) {
+func TestFetchModeFromMetrics(t *testing.T) {
 	testCases := []struct {
 		metrics string
 		mode    import_sstpb.SwitchMode
@@ -149,69 +159,77 @@ func (s *tikvSuite) TestFetchModeFromMetrics(c *C) {
 	}
 
 	for _, tc := range testCases {
-		comment := Commentf("test case '%s'", tc.metrics)
+		comment := fmt.Sprintf("test case '%s'", tc.metrics)
 		mode, err := kv.FetchModeFromMetrics(tc.metrics)
 		if tc.isErr {
-			c.Assert(err, NotNil, comment)
+			require.Error(t, err, comment)
 		} else {
-			c.Assert(err, IsNil, comment)
-			c.Assert(mode, Equals, tc.mode, comment)
+			require.NoError(t, err, comment)
+			require.Equal(t, tc.mode, mode, comment)
 		}
 	}
 }
 
-func (s *tikvSuite) TestCheckPDVersion(c *C) {
+func TestCheckPDVersion(t *testing.T) {
 	var version string
 	ctx := context.Background()
 
 	mockServer := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
-		c.Assert(req.URL.Path, Equals, "/pd/api/v1/version")
+		require.Equal(t, "/pd/api/v1/version", req.URL.Path)
 		w.WriteHeader(http.StatusOK)
 		_, err := w.Write([]byte(version))
-		c.Assert(err, IsNil)
+		require.NoError(t, err)
 	}))
 	mockURL, err := url.Parse(mockServer.URL)
-	c.Assert(err, IsNil)
+	require.NoError(t, err)
 
 	tls := common.NewTLSFromMockServer(mockServer)
 
 	version = `{
     "version": "v4.0.0-rc.2-451-g760fb650"
 }`
-	c.Assert(kv.CheckPDVersion(ctx, tls, mockURL.Host, requiredMinPDVersion, requiredMaxPDVersion), IsNil)
+	require.NoError(t, kv.CheckPDVersion(ctx, tls, mockURL.Host, requiredMinPDVersion, requiredMaxPDVersion))
 
 	version = `{
     "version": "v4.0.0"
 }`
-	c.Assert(kv.CheckPDVersion(ctx, tls, mockURL.Host, requiredMinPDVersion, requiredMaxPDVersion), IsNil)
+	require.NoError(t, kv.CheckPDVersion(ctx, tls, mockURL.Host, requiredMinPDVersion, requiredMaxPDVersion))
 
 	version = `{
     "version": "v9999.0.0"
 }`
-	c.Assert(kv.CheckPDVersion(ctx, tls, mockURL.Host, requiredMinPDVersion, requiredMaxPDVersion), ErrorMatches, "PD version too new.*")
+	err = kv.CheckPDVersion(ctx, tls, mockURL.Host, requiredMinPDVersion, requiredMaxPDVersion)
+	require.Error(t, err)
+	require.Regexp(t, "PD version too new.*", err.Error())
 
 	version = `{
     "version": "v6.0.0"
 }`
-	c.Assert(kv.CheckPDVersion(ctx, tls, mockURL.Host, requiredMinPDVersion, requiredMaxPDVersion), ErrorMatches, "PD version too new.*")
+	err = kv.CheckPDVersion(ctx, tls, mockURL.Host, requiredMinPDVersion, requiredMaxPDVersion)
+	require.Error(t, err)
+	require.Regexp(t, "PD version too new.*", err.Error())
 
 	version = `{
     "version": "v6.0.0-beta"
 }`
-	c.Assert(kv.CheckPDVersion(ctx, tls, mockURL.Host, requiredMinPDVersion, requiredMaxPDVersion), ErrorMatches, "PD version too new.*")
+	err = kv.CheckPDVersion(ctx, tls, mockURL.Host, requiredMinPDVersion, requiredMaxPDVersion)
+	require.Error(t, err)
+	require.Regexp(t, "PD version too new.*", err.Error())
 
 	version = `{
     "version": "v1.0.0"
 }`
-	c.Assert(kv.CheckPDVersion(ctx, tls, mockURL.Host, requiredMinPDVersion, requiredMaxPDVersion), ErrorMatches, "PD version too old.*")
+	err = kv.CheckPDVersion(ctx, tls, mockURL.Host, requiredMinPDVersion, requiredMaxPDVersion)
+	require.Error(t, err)
+	require.Regexp(t, "PD version too old.*", err.Error())
 }
 
-func (s *tikvSuite) TestCheckTiKVVersion(c *C) {
+func TestCheckTiKVVersion(t *testing.T) {
 	var versions []string
 	ctx := context.Background()
 
 	mockServer := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
-		c.Assert(req.URL.Path, Equals, "/pd/api/v1/stores")
+		require.Equal(t, "/pd/api/v1/stores", req.URL.Path)
 		w.WriteHeader(http.StatusOK)
 
 		stores := make([]map[string]interface{}, 0, len(versions))
@@ -227,25 +245,33 @@ func (s *tikvSuite) TestCheckTiKVVersion(c *C) {
 			"count":  len(versions),
 			"stores": stores,
 		})
-		c.Assert(err, IsNil)
+		require.NoError(t, err)
 	}))
 	mockURL, err := url.Parse(mockServer.URL)
-	c.Assert(err, IsNil)
+	require.NoError(t, err)
 
 	tls := common.NewTLSFromMockServer(mockServer)
 
 	versions = []string{"4.1.0", "v4.1.0-alpha-9-ga27a7dd"}
-	c.Assert(kv.CheckTiKVVersion(ctx, tls, mockURL.Host, requiredMinTiKVVersion, requiredMaxTiKVVersion), IsNil)
+	require.Nil(t, kv.CheckTiKVVersion(ctx, tls, mockURL.Host, requiredMinTiKVVersion, requiredMaxTiKVVersion))
 
 	versions = []string{"9999.0.0", "4.0.0"}
-	c.Assert(kv.CheckTiKVVersion(ctx, tls, mockURL.Host, requiredMinTiKVVersion, requiredMaxTiKVVersion), ErrorMatches, `TiKV \(at tikv0\.test:20160\) version too new.*`)
+	err = kv.CheckPDVersion(ctx, tls, mockURL.Host, requiredMinPDVersion, requiredMaxPDVersion)
+	require.Error(t, err)
+	require.Regexp(t, `TiKV \(at tikv0\.test:20160\) version too new.*`, err.Error())
 
 	versions = []string{"4.0.0", "1.0.0"}
-	c.Assert(kv.CheckTiKVVersion(ctx, tls, mockURL.Host, requiredMinTiKVVersion, requiredMaxTiKVVersion), ErrorMatches, `TiKV \(at tikv1\.test:20160\) version too old.*`)
+	err = kv.CheckPDVersion(ctx, tls, mockURL.Host, requiredMinPDVersion, requiredMaxPDVersion)
+	require.Error(t, err)
+	require.Regexp(t, `TiKV \(at tikv0\.test:20160\) version too old.*`, err.Error())
 
 	versions = []string{"6.0.0"}
-	c.Assert(kv.CheckTiKVVersion(ctx, tls, mockURL.Host, requiredMinTiKVVersion, requiredMaxTiKVVersion), ErrorMatches, `TiKV \(at tikv0\.test:20160\) version too new.*`)
+	err = kv.CheckPDVersion(ctx, tls, mockURL.Host, requiredMinPDVersion, requiredMaxPDVersion)
+	require.Error(t, err)
+	require.Regexp(t, `TiKV \(at tikv0\.test:20160\) version too new.*`, err.Error())
 
 	versions = []string{"6.0.0-beta"}
-	c.Assert(kv.CheckTiKVVersion(ctx, tls, mockURL.Host, requiredMinTiKVVersion, requiredMaxTiKVVersion), ErrorMatches, `TiKV \(at tikv0\.test:20160\) version too new.*`)
+	err = kv.CheckPDVersion(ctx, tls, mockURL.Host, requiredMinPDVersion, requiredMaxPDVersion)
+	require.Error(t, err)
+	require.Regexp(t, `TiKV \(at tikv0\.test:20160\) version too new.*`, err.Error())
 }
```

</details>

<details>
<summary>the failures</summary>

```
=== RUN   TestCheckTiKVVersion
    tikv_test.go:218: 
        	Error Trace:	tikv_test.go:218
        	            				server.go:2047
        	            				server.go:2879
        	            				server.go:1930
        	            				asm_amd64.s:1581
        	Error:      	Not equal: 
        	            	expected: "/pd/api/v1/stores"
        	            	actual  : "/pd/api/v1/version"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-/pd/api/v1/stores
        	            	+/pd/api/v1/version
        	Test:       	TestCheckTiKVVersion
    tikv_test.go:218: 
        	Error Trace:	tikv_test.go:218
        	            				server.go:2047
        	            				server.go:2879
        	            				server.go:1930
        	            				asm_amd64.s:1581
        	Error:      	Not equal: 
        	            	expected: "/pd/api/v1/stores"
        	            	actual  : "/pd/api/v1/version"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-/pd/api/v1/stores
        	            	+/pd/api/v1/version
        	Test:       	TestCheckTiKVVersion
    tikv_test.go:247: 
        	Error Trace:	tikv_test.go:247
        	Error:      	Expect "Get "https://127.0.0.1:49317/pd/api/v1/version": EOF" to match "TiKV \(at tikv0\.test:20160\) version too new.*"
        	Test:       	TestCheckTiKVVersion
--- FAIL: TestCheckTiKVVersion (0.00s)


Expected :/pd/api/v1/stores
Actual   :/pd/api/v1/version
```

</details>

--- a/br/pkg/lightning/metric/metric_test.go
+++ b/br/pkg/lightning/metric/metric_test.go
@@ -18,43 +18,32 @@ import (
 	"errors"
 	"testing"
 
-	. "github.com/pingcap/check"
 	"github.com/pingcap/tidb/br/pkg/lightning/metric"
 	"github.com/prometheus/client_golang/prometheus"
+	"github.com/stretchr/testify/require"
 )
 
-type testMetricSuite struct{}
-
-func (s *testMetricSuite) SetUpSuite(c *C)    {}
-func (s *testMetricSuite) TearDownSuite(c *C) {}
-
-var _ = Suite(&testMetricSuite{})
-
-func TestMetric(t *testing.T) {
-	TestingT(t)
-}
-
-func (s *testMetricSuite) TestReadCounter(c *C) {
+func TestReadCounter(t *testing.T) {
 	counter := prometheus.NewCounter(prometheus.CounterOpts{})
 	counter.Add(1256.0)
 	counter.Add(2214.0)
-	c.Assert(metric.ReadCounter(counter), Equals, 3470.0)
+	require.Equal(t, 3470.0, metric.ReadCounter(counter))
 }
 
-func (s *testMetricSuite) TestReadHistogramSum(c *C) {
+func TestReadHistogramSum(t *testing.T) {
 	histogram := prometheus.NewHistogram(prometheus.HistogramOpts{})
 	histogram.Observe(11131.5)
 	histogram.Observe(15261.0)
-	c.Assert(metric.ReadHistogramSum(histogram), Equals, 26392.5)
+	require.Equal(t, 26392.5, metric.ReadHistogramSum(histogram))
 }
 
-func (s *testMetricSuite) TestRecordEngineCount(c *C) {
+func TestRecordEngineCount(t *testing.T) {
 	metric.RecordEngineCount("table1", nil)
 	metric.RecordEngineCount("table1", errors.New("mock error"))
 	successCounter, err := metric.ProcessedEngineCounter.GetMetricWithLabelValues("table1", "success")
-	c.Assert(err, IsNil)
-	c.Assert(metric.ReadCounter(successCounter), Equals, 1.0)
+	require.NoError(t, err)
+	require.Equal(t, 1.0, metric.ReadCounter(successCounter))
 	failureCount, err := metric.ProcessedEngineCounter.GetMetricWithLabelValues("table1", "failure")
-	c.Assert(err, IsNil)
-	c.Assert(metric.ReadCounter(failureCount), Equals, 1.0)
+	require.NoError(t, err)
+	require.Equal(t, 1.0, metric.ReadCounter(failureCount))
 }

--- a/br/pkg/lightning/tikv/tikv_test.go
+++ b/br/pkg/lightning/tikv/tikv_test.go
@@ -1,3 +1,17 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
 package tikv_test
 
 import (
@@ -9,17 +23,14 @@ import (
 	"net/url"
 	"sort"
 	"sync"
+	"testing"
 
 	"github.com/coreos/go-semver/semver"
-	. "github.com/pingcap/check"
 	"github.com/pingcap/kvproto/pkg/import_sstpb"
 	"github.com/pingcap/tidb/br/pkg/lightning/common"
 	kv "github.com/pingcap/tidb/br/pkg/lightning/tikv"
+	"github.com/stretchr/testify/require"
 )
-
-type tikvSuite struct{}
-
-var _ = Suite(&tikvSuite{})
 
 var (
 	// Samples from importer backend for testing the Check***Version functions.
@@ -30,7 +41,7 @@ var (
 	requiredMaxTiKVVersion = *semver.New("6.0.0")
 )
 
-func (s *tikvSuite) TestForAllStores(c *C) {
+func TestForAllStores(t *testing.T) {
 	server := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
 		_, err := w.Write([]byte(`
 			{
@@ -84,7 +95,7 @@ func (s *tikvSuite) TestForAllStores(c *C) {
 				]
 			}
 		`))
-		c.Assert(err, IsNil)
+		require.NoError(t, err)
 	}))
 	defer server.Close()
 
@@ -100,11 +111,10 @@ func (s *tikvSuite) TestForAllStores(c *C) {
 		allStoresLock.Unlock()
 		return nil
 	})
-	c.Assert(err, IsNil)
+	require.NoError(t, err)
 
 	sort.Slice(allStores, func(i, j int) bool { return allStores[i].Address < allStores[j].Address })
-
-	c.Assert(allStores, DeepEquals, []*kv.Store{
+	require.Equal(t, []*kv.Store{
 		{
 			Address: "127.0.0.1:20160",
 			Version: "3.0.0-beta.1",
@@ -125,10 +135,10 @@ func (s *tikvSuite) TestForAllStores(c *C) {
 			Version: "3.0.1",
 			State:   kv.StoreStateOffline,
 		},
-	})
+	}, allStores)
 }
 
-func (s *tikvSuite) TestFetchModeFromMetrics(c *C) {
+func TestFetchModeFromMetrics(t *testing.T) {
 	testCases := []struct {
 		metrics string
 		mode    import_sstpb.SwitchMode
@@ -149,69 +159,77 @@ func (s *tikvSuite) TestFetchModeFromMetrics(c *C) {
 	}
 
 	for _, tc := range testCases {
-		comment := Commentf("test case '%s'", tc.metrics)
+		comment := fmt.Sprintf("test case '%s'", tc.metrics)
 		mode, err := kv.FetchModeFromMetrics(tc.metrics)
 		if tc.isErr {
-			c.Assert(err, NotNil, comment)
+			require.Error(t, err, comment)
 		} else {
-			c.Assert(err, IsNil, comment)
-			c.Assert(mode, Equals, tc.mode, comment)
+			require.NoError(t, err, comment)
+			require.Equal(t, tc.mode, mode, comment)
 		}
 	}
 }
 
-func (s *tikvSuite) TestCheckPDVersion(c *C) {
+func TestCheckPDVersion(t *testing.T) {
 	var version string
 	ctx := context.Background()
 
 	mockServer := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
-		c.Assert(req.URL.Path, Equals, "/pd/api/v1/version")
+		require.Equal(t, "/pd/api/v1/version", req.URL.Path)
 		w.WriteHeader(http.StatusOK)
 		_, err := w.Write([]byte(version))
-		c.Assert(err, IsNil)
+		require.NoError(t, err)
 	}))
 	mockURL, err := url.Parse(mockServer.URL)
-	c.Assert(err, IsNil)
+	require.NoError(t, err)
 
 	tls := common.NewTLSFromMockServer(mockServer)
 
 	version = `{
     "version": "v4.0.0-rc.2-451-g760fb650"
 }`
-	c.Assert(kv.CheckPDVersion(ctx, tls, mockURL.Host, requiredMinPDVersion, requiredMaxPDVersion), IsNil)
+	require.NoError(t, kv.CheckPDVersion(ctx, tls, mockURL.Host, requiredMinPDVersion, requiredMaxPDVersion))
 
 	version = `{
     "version": "v4.0.0"
 }`
-	c.Assert(kv.CheckPDVersion(ctx, tls, mockURL.Host, requiredMinPDVersion, requiredMaxPDVersion), IsNil)
+	require.NoError(t, kv.CheckPDVersion(ctx, tls, mockURL.Host, requiredMinPDVersion, requiredMaxPDVersion))
 
 	version = `{
     "version": "v9999.0.0"
 }`
-	c.Assert(kv.CheckPDVersion(ctx, tls, mockURL.Host, requiredMinPDVersion, requiredMaxPDVersion), ErrorMatches, "PD version too new.*")
+	err = kv.CheckPDVersion(ctx, tls, mockURL.Host, requiredMinPDVersion, requiredMaxPDVersion)
+	require.Error(t, err)
+	require.Regexp(t, "PD version too new.*", err.Error())
 
 	version = `{
     "version": "v6.0.0"
 }`
-	c.Assert(kv.CheckPDVersion(ctx, tls, mockURL.Host, requiredMinPDVersion, requiredMaxPDVersion), ErrorMatches, "PD version too new.*")
+	err = kv.CheckPDVersion(ctx, tls, mockURL.Host, requiredMinPDVersion, requiredMaxPDVersion)
+	require.Error(t, err)
+	require.Regexp(t, "PD version too new.*", err.Error())
 
 	version = `{
     "version": "v6.0.0-beta"
 }`
-	c.Assert(kv.CheckPDVersion(ctx, tls, mockURL.Host, requiredMinPDVersion, requiredMaxPDVersion), ErrorMatches, "PD version too new.*")
+	err = kv.CheckPDVersion(ctx, tls, mockURL.Host, requiredMinPDVersion, requiredMaxPDVersion)
+	require.Error(t, err)
+	require.Regexp(t, "PD version too new.*", err.Error())
 
 	version = `{
     "version": "v1.0.0"
 }`
-	c.Assert(kv.CheckPDVersion(ctx, tls, mockURL.Host, requiredMinPDVersion, requiredMaxPDVersion), ErrorMatches, "PD version too old.*")
+	err = kv.CheckPDVersion(ctx, tls, mockURL.Host, requiredMinPDVersion, requiredMaxPDVersion)
+	require.Error(t, err)
+	require.Regexp(t, "PD version too old.*", err.Error())
 }
 
-func (s *tikvSuite) TestCheckTiKVVersion(c *C) {
+func TestCheckTiKVVersion(t *testing.T) {
 	var versions []string
 	ctx := context.Background()
 
 	mockServer := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
-		c.Assert(req.URL.Path, Equals, "/pd/api/v1/stores")
+		require.Equal(t, "/pd/api/v1/stores", req.URL.Path)
 		w.WriteHeader(http.StatusOK)
 
 		stores := make([]map[string]interface{}, 0, len(versions))
@@ -227,25 +245,33 @@ func (s *tikvSuite) TestCheckTiKVVersion(c *C) {
 			"count":  len(versions),
 			"stores": stores,
 		})
-		c.Assert(err, IsNil)
+		require.NoError(t, err)
 	}))
 	mockURL, err := url.Parse(mockServer.URL)
-	c.Assert(err, IsNil)
+	require.NoError(t, err)
 
 	tls := common.NewTLSFromMockServer(mockServer)
 
 	versions = []string{"4.1.0", "v4.1.0-alpha-9-ga27a7dd"}
-	c.Assert(kv.CheckTiKVVersion(ctx, tls, mockURL.Host, requiredMinTiKVVersion, requiredMaxTiKVVersion), IsNil)
+	require.NoError(t, kv.CheckTiKVVersion(ctx, tls, mockURL.Host, requiredMinTiKVVersion, requiredMaxTiKVVersion))
 
 	versions = []string{"9999.0.0", "4.0.0"}
-	c.Assert(kv.CheckTiKVVersion(ctx, tls, mockURL.Host, requiredMinTiKVVersion, requiredMaxTiKVVersion), ErrorMatches, `TiKV \(at tikv0\.test:20160\) version too new.*`)
+	err = kv.CheckTiKVVersion(ctx, tls, mockURL.Host, requiredMinPDVersion, requiredMaxPDVersion)
+	require.Error(t, err)
+	require.Regexp(t, `TiKV \(at tikv0\.test:20160\) version too new.*`, err.Error())
 
 	versions = []string{"4.0.0", "1.0.0"}
-	c.Assert(kv.CheckTiKVVersion(ctx, tls, mockURL.Host, requiredMinTiKVVersion, requiredMaxTiKVVersion), ErrorMatches, `TiKV \(at tikv1\.test:20160\) version too old.*`)
+	err = kv.CheckTiKVVersion(ctx, tls, mockURL.Host, requiredMinPDVersion, requiredMaxPDVersion)
+	require.Error(t, err)
+	require.Regexp(t, `TiKV \(at tikv1\.test:20160\) version too old.*`, err.Error())
 
 	versions = []string{"6.0.0"}
-	c.Assert(kv.CheckTiKVVersion(ctx, tls, mockURL.Host, requiredMinTiKVVersion, requiredMaxTiKVVersion), ErrorMatches, `TiKV \(at tikv0\.test:20160\) version too new.*`)
+	err = kv.CheckTiKVVersion(ctx, tls, mockURL.Host, requiredMinPDVersion, requiredMaxPDVersion)
+	require.Error(t, err)
+	require.Regexp(t, `TiKV \(at tikv0\.test:20160\) version too new.*`, err.Error())
 
 	versions = []string{"6.0.0-beta"}
-	c.Assert(kv.CheckTiKVVersion(ctx, tls, mockURL.Host, requiredMinTiKVVersion, requiredMaxTiKVVersion), ErrorMatches, `TiKV \(at tikv0\.test:20160\) version too new.*`)
+	err = kv.CheckTiKVVersion(ctx, tls, mockURL.Host, requiredMinPDVersion, requiredMaxPDVersion)
+	require.Error(t, err)
+	require.Regexp(t, `TiKV \(at tikv0\.test:20160\) version too new.*`, err.Error())
 }

--- a/br/pkg/lightning/verification/checksum_test.go
+++ b/br/pkg/lightning/verification/checksum_test.go
@@ -18,34 +18,23 @@ import (
 	"encoding/json"
 	"testing"
 
-	. "github.com/pingcap/check"
 	"github.com/pingcap/tidb/br/pkg/lightning/common"
 	"github.com/pingcap/tidb/br/pkg/lightning/verification"
+	"github.com/stretchr/testify/require"
 )
-
-type testKVChcksumSuite struct{}
-
-func (s *testKVChcksumSuite) SetUpSuite(c *C)    {}
-func (s *testKVChcksumSuite) TearDownSuite(c *C) {}
-
-var _ = Suite(&testKVChcksumSuite{})
-
-func TestKVChcksum(t *testing.T) {
-	TestingT(t)
-}
 
 func uint64NotEqual(a uint64, b uint64) bool { return a != b }
 
-func (s *testKVChcksumSuite) TestChcksum(c *C) {
+func TestChcksum(t *testing.T) {
 	checksum := verification.NewKVChecksum(0)
-	c.Assert(checksum.Sum(), Equals, uint64(0))
+	require.Equal(t, uint64(0), checksum.Sum())
 
 	// checksum on nothing
 	checksum.Update([]common.KvPair{})
-	c.Assert(checksum.Sum(), Equals, uint64(0))
+	require.Equal(t, uint64(0), checksum.Sum())
 
 	checksum.Update(nil)
-	c.Assert(checksum.Sum(), Equals, uint64(0))
+	require.Equal(t, uint64(0), checksum.Sum())
 
 	// checksum on real data
 	excpectChecksum := uint64(4850203904608948940)
@@ -67,18 +56,18 @@ func (s *testKVChcksumSuite) TestChcksum(c *C) {
 	for _, kv := range kvs {
 		kvBytes += uint64(len(kv.Key) + len(kv.Val))
 	}
-	c.Assert(checksum.SumSize(), Equals, kvBytes)
-	c.Assert(checksum.SumKVS(), Equals, uint64(len(kvs)))
-	c.Assert(checksum.Sum(), Equals, excpectChecksum)
+	require.Equal(t, kvBytes, checksum.SumSize())
+	require.Equal(t, uint64(len(kvs)), checksum.SumKVS())
+	require.Equal(t, excpectChecksum, checksum.Sum())
 
 	// recompute on same key-value
 	checksum.Update(kvs)
-	c.Assert(checksum.SumSize(), Equals, kvBytes<<1)
-	c.Assert(checksum.SumKVS(), Equals, uint64(len(kvs))<<1)
-	c.Assert(uint64NotEqual(checksum.Sum(), excpectChecksum), IsTrue)
+	require.Equal(t, kvBytes<<1, checksum.SumSize())
+	require.Equal(t, uint64(len(kvs))<<1, checksum.SumKVS())
+	require.True(t, uint64NotEqual(checksum.Sum(), excpectChecksum))
 }
 
-func (s *testKVChcksumSuite) TestChecksumJSON(c *C) {
+func TestChecksumJSON(t *testing.T) {
 	testStruct := &struct {
 		Checksum verification.KVChecksum
 	}{
@@ -87,6 +76,6 @@ func (s *testKVChcksumSuite) TestChecksumJSON(c *C) {
 
 	res, err := json.Marshal(testStruct)
 
-	c.Assert(err, IsNil)
-	c.Assert(res, BytesEquals, []byte(`{"Checksum":{"checksum":7890,"size":123,"kvs":456}}`))
+	require.NoError(t, err)
+	require.Equal(t, []byte(`{"Checksum":{"checksum":7890,"size":123,"kvs":456}}`), res)
 }

--- a/br/pkg/lightning/worker/worker_test.go
+++ b/br/pkg/lightning/worker/worker_test.go
@@ -18,39 +18,28 @@ import (
 	"context"
 	"testing"
 
-	. "github.com/pingcap/check"
 	"github.com/pingcap/tidb/br/pkg/lightning/worker"
+	"github.com/stretchr/testify/require"
 )
 
-type testWorkerPool struct{}
-
-func (s *testWorkerPool) SetUpSuite(c *C)    {}
-func (s *testWorkerPool) TearDownSuite(c *C) {}
-
-var _ = Suite(&testWorkerPool{})
-
-func TestNewRestoreWorkerPool(t *testing.T) {
-	TestingT(t)
-}
-
-func (s *testWorkerPool) TestApplyRecycle(c *C) {
+func TestApplyRecycle(t *testing.T) {
 	pool := worker.NewPool(context.Background(), 3, "test")
 
 	w1, w2, w3 := pool.Apply(), pool.Apply(), pool.Apply()
-	c.Assert(w1.ID, Equals, int64(1))
-	c.Assert(w2.ID, Equals, int64(2))
-	c.Assert(w3.ID, Equals, int64(3))
-	c.Assert(pool.HasWorker(), Equals, false)
+	require.Equal(t, int64(1), w1.ID)
+	require.Equal(t, int64(2), w2.ID)
+	require.Equal(t, int64(3), w3.ID)
+	require.Equal(t, false, pool.HasWorker())
 
 	pool.Recycle(w3)
-	c.Assert(pool.HasWorker(), Equals, true)
-	c.Assert(pool.Apply(), Equals, w3)
+	require.Equal(t, true, pool.HasWorker())
+	require.Equal(t, w3, pool.Apply())
 	pool.Recycle(w2)
-	c.Assert(pool.Apply(), Equals, w2)
+	require.Equal(t, w2, pool.Apply())
 	pool.Recycle(w1)
-	c.Assert(pool.Apply(), Equals, w1)
+	require.Equal(t, w1, pool.Apply())
 
-	c.Assert(pool.HasWorker(), Equals, false)
+	require.Equal(t, false, pool.HasWorker())
 
-	c.Assert(func() { pool.Recycle(nil) }, PanicMatches, "invalid restore worker")
+	require.PanicsWithValue(t, "invalid restore worker", func() { pool.Recycle(nil) })
 }

